src/traces/heatmap/calc.js

```diff
@@ -64,14 +64,17 @@ module.exports = function calc(gd, trace) {
         } else {
             x = trace.x ? xa.makeCalcdata(trace, 'x') : [];
             y = trace.y ? ya.makeCalcdata(trace, 'y') : [];
+
+            trace._x = x;
+            trace._y = y;
```

Comment:
Minor style tip: we could merge this with the above statements and have:

```js
x = trace._x = trace.x ? xa.makeCalcdata(trace, 'x') : [];
// similarly for y
```

Reply:
Done in 59edb5704f076b438e6b8460b3e03097d61748ab